axum/src/routing/mod.rs

```diff
@@ -506,6 +500,16 @@ where
             Endpoint::NestedRouter(router) => router.call_with_state(req, state),
         }
     }
+
+    fn next_route_id(&mut self) -> RouteId {
+        let next_id = self
+            .prev_route_id
+            .0
+            .checked_add(1)
+            .expect("Over `u32::MAX` routes created. If you need this, please file an issue.");
+        self.prev_route_id = RouteId(next_id);
+        self.prev_route_id
```

Comment:
I still we should name this `next_route_id` and return `RouteId(0)` as the first one as before, not `RouteId(1)`. But that's very nitpicky at this point.

Reply:
Hehe I think I'll just leave it like this.